rpc/apis.go

```diff
@@ -33,7 +34,7 @@ func GetRPCAPIs(cliCtx context.CLIContext, key emintcrypto.PrivKeySecp256k1) []r
 		{
 			Namespace: "eth",
 			Version:   "1.0",
-			Service:   NewPublicFilterAPI(cliCtx),
+			Service:   NewPublicFilterAPI(cliCtx, ethAPI),
```

Comment:
Why use the other API as the backend to the filter API? Shouldn't the backend be common for both instead of all requests going through eth API? 

Might make sense for now but when you get into needing the filter specific functions, you would then have to implement it within the eth API

Reply:
I was thinking that we use the ethAPI functions to poll for changes for now, otherwise we would have to re-implement certain functions like eth_getBlockByNumber if we just use the context directly. I could change it though

Reply:
wouldn't have to reimplement, but would require moving that logic, in this case `getBlockByNumber`, to that backend type which is used by both the eth and filter API. Just so the backend logic isn't tied to the eth api to make it more easily extensible, since it will inevitably have to change to this if you expand and rely on this interface

Reply:
that would make a lot more sense🤔 will update